fdbserver/ConsistencyScan.actor.cpp

```diff
@@ -41,31 +46,896 @@
+	loop {
+		try {
+			SystemDBWriteLockedNow(db.getReference())->setOptions(tr);
+			state ConsistencyScanState::Config config = wait(cs.config().getD(tr));
+
+			// Sometimes enable if disabled, otherwise sometimes update the scan min version to restart it
+			if (!config.enabled &&
+			    g_simulator->consistencyScanState == ISimulator::SimConsistencyScanState::DisabledStart &&
+			    deterministicRandom()->random01() < 0.5) {
+				// just don't enable it!
+				TraceEvent("ConsistencyScan_SimEnableSkip", csId).log();
+				return Void();
+			}
+			if (!config.enabled && g_simulator->consistencyScanState < ISimulator::SimConsistencyScanState::Enabled) {
+				g_simulator->consistencyScanState = ISimulator::SimConsistencyScanState::Enabling;
+				config.enabled = true;
+			} else if (BUGGIFY_WITH_PROB(0.5)) {
+				config.minStartVersion = tr->getReadVersion().get();
+			}
+			// also change the rate
+			config.maxReadByteRate = deterministicRandom()->randomInt(1, 50e6);
+			config.targetRoundTimeSeconds = deterministicRandom()->randomSkewedUInt32(1, 200);
+			config.minRoundTimeSeconds = deterministicRandom()->randomSkewedUInt32(1, 200);
+
+			// TODO:  Reconfigure cs.rangeConfig() randomly
+
+			if (config.enabled) {
+				cs.config().set(tr, config);
+				wait(tr->commit());
+
+				g_simulator->consistencyScanState = ISimulator::SimConsistencyScanState::Enabled;
+				TraceEvent("ConsistencyScan_SimEnabled")
+				    .detail("MaxReadByteRate", config.maxReadByteRate)
+				    .detail("TargetRoundTimeSeconds", config.targetRoundTimeSeconds)
+				    .detail("MinRoundTimeSeconds", config.minRoundTimeSeconds);
+				CODE_PROBE(true, "Consistency Scan enabled in simulation");
+			}
+
+			break;
+		} catch (Error& e) {
+			wait(tr->onError(e));
+		}
+	}
+
+	return Void();
+}
+
+ACTOR Future<Void> disableConsistencyScanInSim(Database db, ConsistencyScanMemoryState* memState) {
+	state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(db);
+	state ConsistencyScanState cs;
+	TraceEvent("ConsistencyScan_SimDisableWaiting", memState->csId).log();
+	// Also wait until we've scanned at least one round by checking stats
+	state bool waitForRoundComplete = true;
+	// FIXME: also wait until we've done the canary failure
+	ASSERT(g_network->isSimulated());
+	loop {
+		if (g_simulator->speedUpSimulation &&
+		    g_simulator->consistencyScanState >= ISimulator::SimConsistencyScanState::Enabled) {
+			break;
+		}
+		double delayTime = std::max(1.0, FLOW_KNOBS->SIM_SPEEDUP_AFTER_SECONDS - now());
+		wait(delay(delayTime));
+	}
+	TraceEvent("ConsistencyScan_SimDisable", memState->csId).log();
+	loop {
+		try {
+			SystemDBWriteLockedNow(db.getReference())->setOptions(tr);
+			state ConsistencyScanState::Config config = wait(cs.config().getD(tr));
+			state bool skipDisable = false;
+			// see if any rounds have completed
+			ConsistencyScanState::StatsHistoryMap::RangeResultType olderStats =
```

Comment:
Figure out effect on test runtime/cpu of waiting for this.
If it's non-trivial, maybe sometimes don't enable this check